kafka/pykafka/cluster.py

```diff
@@ -95,6 +98,38 @@ def _update_topics(self, metadata):
             else:
                 self._topics[name].update(meta)
 
+    def get_offset_manager(self, consumer_group_name):
+        """Get the broker designated as the offset manager for this consumer
+            group
+
+        Based on Step 1 at https://cwiki.apache.org/confluence/display/KAFKA/Committing+and+fetching+consumer+offsets+in+Kafka
+
+        :param consumer_group_name: the name of the consumer group
+        :type consumer_group_name: str
+        """
+        # arbitrarily choose a broker, since this request can go to any
+        broker = self.brokers[random.choice(self.brokers.keys())]
+        backoff, retries = 2, 0
+        MAX_RETRIES = 3
+        while True:
+            try:
+                retries += 1
+                req = ConsumerMetadataRequest(consumer_group_name)
+                future = broker.handler.request(req)
+                res = future.get(ConsumerMetadataResponse)
+            except Exception:
+                logger.debug('Error discovering offset manager. Sleeping for {}s'.format(backoff))
+                if retries < MAX_RETRIES:
+                    time.sleep(backoff)  # XXX - not sure if this works here
```

Comment:
When I wrote the comment, I was concerned about how this function would be used with respect to threading and if there was some existing backoff functionality I should be reusing.